core/modules/loader/ServerUdpBase.cc

```diff
@@ -0,0 +1,152 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+// Class header
+#include "loader/ServerUdpBase.h"
+
+// System headers
+#include <iostream>
+#include <unistd.h>
+
+// Third-party headers
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.loader.ServerUdpBase");
+}
+
+namespace lsst {
+namespace qserv {
+namespace loader {
+
+std::atomic<uint64_t> ServerUdpBase::_msgIdSeq{1};
+
+
+ServerUdpBase::ServerUdpBase(boost::asio::io_service& io_service, std::string const& host, int port)
+    : _ioService(io_service),
+      _socket(io_service, boost::asio::ip::udp::endpoint(boost::asio::ip::udp::v4(), port)),
```

Comment:
This particular socket constructor opens and binds the socket, and could fail under several likely scenarios (port already in use for example).  Those will cause a throw from your constructor.  That's okay if you've accounted for that possibility?